x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/ClusterPermission.java

```diff
@@ -145,6 +155,14 @@ private static Automaton createAutomaton(Set<String> allowedActionPatterns, Set<
          */
         boolean check(String action, TransportRequest request, Authentication authentication);
 
+        /**
+         * Checks permission to a cluster action regardless of the request and authentication context.
+         *
+         * @param action  action name
+         * @return {@code true} if the specified action execution can be granted by given permission else returns {@code false}
+         */
+        boolean check(String action);
```

Comment:
I think I understand why you added this; you need to verify that some privileges match an action, but there are privileges (solely `manage_own_api_keys` is a named privilege like this, right?) that need to look at more than just the action to tell. In this case, the check is imperfect because the privilege might not actually grant the action. Am I right?

In case I am, I would squarely classify this as a hack. Outside of audit filtering by privileges we would never call this method.
If we're introducing a hack it should be localized (eg in `LoggingAuditTrail`). I don't like that it extends to a permission that is used in a role for authorization.